lib/rex/post/meterpreter/packet_dispatcher.rb

```diff
@@ -639,6 +638,44 @@ def deregister_inbound_handler(handler)
   attr_accessor :receiver_thread # :nodoc:
   attr_accessor :dispatcher_thread # :nodoc:
   attr_accessor :waiters # :nodoc:
+
+  @tlv_log_file = nil
+
+  def log_packet(packet, packet_type)
+    if framework.datastore['sessiontlvlogging']&.downcase == 'console'
+      log_packet_to_console(packet, packet_type)
+    elsif framework.datastore['sessiontlvlogging']&.downcase =~ /^file:.+/
+      log_packet_to_file(packet, packet_type)
+    end
+  end
+
+  def log_packet_to_console(packet, packet_type)
+    if packet_type == :inbound
+      print "\n%bluRECV%clr: #{packet.inspect}\n"
+    elsif packet_type == :outbound
+      print "\n%redSEND%clr: #{packet.inspect}\n"
+    end
+  end
+
+  def log_packet_to_file(packet, packet_type)
+    path = framework.datastore['sessiontlvlogging'].downcase.split('file:').last
+    pathname = ::Pathname.new(path)
+    if pathname.directory?
+      if @tlv_log_file.nil?
+        @tlv_log_file = ::File.open(::File.dirname(pathname) + '/sessiontlvlogging.txt', 'a+')
+      end
+    elsif pathname.file? || !pathname.exist?
+      if @tlv_log_file.nil?
+        @tlv_log_file = ::File.open(pathname, 'a+')
```

Comment:
Do we want to close/set this file explicitly at some point? Potentially with:

```
  #
  # Cleans up the meterpreter instance, terminating the dispatcher thread.
  #
  def cleanup_meterpreter
      tlv_log_file.close if tlv_log_file
```

With the corresponding attr_accessor

```
  attr_accessor :receiver_thread # :nodoc:
  attr_accessor :dispatcher_thread # :nodoc:
  attr_accessor :tlv_log_file# :nodoc:
  attr_accessor :waiters # :nodoc:
```

Disclaimer: No idea if that will work, if it's a good idea, or how it behaves on windows